tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -0,0 +1,482 @@
+
+    int result = bpf_prog_attach(
+        bpf_program__fd(const_cast<const bpf_program*>(connect_program_v4)), 0, BPF_CGROUP_INET4_CONNECT, 0);
+    REQUIRE(result == 0);
+
+    bpf_program* connect_program_v6 = bpf_object__find_program_by_name(object, "authorize_connect6");
+    REQUIRE(connect_program_v6 != nullptr);
+
+    result = bpf_prog_attach(
+        bpf_program__fd(const_cast<const bpf_program*>(connect_program_v6)), 0, BPF_CGROUP_INET6_CONNECT, 0);
+    REQUIRE(result == 0);
+
+    *return_object = object;
+}
+
+static void
+_update_policy_map(
+    _In_ const struct bpf_object* object,
+    _In_ sockaddr_storage& destination,
+    _In_ sockaddr_storage& proxy,
+    uint16_t port,
+    uint32_t protocol,
+    bool dual_stack,
+    bool add)
+{
+    bpf_map* policy_map = bpf_object__find_map_by_name(object, "policy_map");
+    REQUIRE(policy_map != nullptr);
+
+    fd_t map_fd = bpf_map__fd(policy_map);
+
+    // Insert / delete redirect policy entry in the map.
+    destination_entry_t key = {0};
+    destination_entry_t value = {0};
+
+    if (_globals.family == AF_INET) {
+        const uint8_t* v4_destination = nullptr;
+        const uint8_t* v4_proxy = nullptr;
+
+        if (dual_stack) {
+            struct sockaddr_in6* v6_destination = (struct sockaddr_in6*)&destination;
+            struct sockaddr_in6* v6_proxy = (struct sockaddr_in6*)&proxy;
+
+            v4_destination = IN6_GET_ADDR_V4MAPPED((IN6_ADDR*)&v6_destination->sin6_addr);
+            v4_proxy = IN6_GET_ADDR_V4MAPPED((IN6_ADDR*)&v6_proxy->sin6_addr);
+        } else {
+            v4_destination = (UCHAR*)&((struct sockaddr_in*)&destination)->sin_addr.S_un.S_addr;
+            v4_proxy = (UCHAR*)&((struct sockaddr_in*)&proxy)->sin_addr.S_un.S_addr;
+        }
+
+        key.destination_ip.ipv4 = *((uint32_t*)v4_destination);
+        value.destination_ip.ipv4 = *((uint32_t*)v4_proxy);
+    } else {
+        struct sockaddr_in6* v6_destination = (struct sockaddr_in6*)&destination;
+        struct sockaddr_in6* v6_proxy = (struct sockaddr_in6*)&proxy;
+        memcpy(key.destination_ip.ipv6, v6_destination->sin6_addr.u.Byte, sizeof(key.destination_ip.ipv6));
+        memcpy(value.destination_ip.ipv6, v6_proxy->sin6_addr.u.Byte, sizeof(value.destination_ip.ipv6));
+    }
+
+    key.destination_port = value.destination_port = htons(port);
+    key.protocol = protocol;
+
+    if (add) {
+        REQUIRE(bpf_map_update_elem(map_fd, &key, &value, 0) == 0);
+    } else {
+        REQUIRE(bpf_map_delete_elem(map_fd, &key) == 0);
+    }
+}
+
+void
+connect_redirect_test(
+    _In_ const struct bpf_object* object,
+    _In_ sender_socket_t* sender_socket,
+    _In_ sockaddr_storage& destination,
+    _In_ sockaddr_storage& proxy,
+    bool dual_stack)
+{
+    // Update policy in the map to redirect the connection to the proxy.
+    _update_policy_map(object, destination, proxy, _globals.remote_port, _globals.protocol, dual_stack, true);
```

Comment:
bool add; instead of literal true as the last param? For readability.

Reply:
updated.